### PR TITLE
Added Engine Disposal Documentation

### LINKED ISSUE #23086: It is not clear whether database connection need to be closed manually after calling pd.read_sql()
The document does not mention it.

### PR DESCRIPTION
**Pull request by poojahagawane**
Created hyperlink to SQLAlchemy docs for Engine Disposal
responsibilities of user. Unified wording of legacy support for sqlite

- [x] closes #23086 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/io/sql.py [hunk 1]
```diff
@@ -140,10 +140,14 @@ def execute(sql, con, cur=None, params=None):
...
         SQL query to be executed.
-    con : SQLAlchemy connectable(engine/connection) or sqlite3 connection
-        Using SQLAlchemy makes it possible to use any DB supported by the
-        library.
-        If a DBAPI2 object, only sqlite3 is supported.
+    con : SQLAlchemy connectable (engine/connection) or database string URI
+        or DBAPI2 connection (fallback mode)
+        Using SQLAlchemy makes it possible to use any DB supported by that
+        library. Legacy support is provided for sqlite3.Connection objects.
+        Note: The user is responsible for engine disposal and connection
```
Comment (gfyoung) [inline pandas/io/sql.py]: Newline above this

Comment (gfyoung) [inline pandas/io/sql.py]:
```suggestion

        Note: The user is responsible for engine disposal and connection
```

Comment (gfyoung) [reply pandas/io/sql.py]: For these "Note:" sections, let's add a newline above (this applies to your other changes)

File: doc/source/whatsnew/v0.25.1.rst [hunk 2]
```diff
@@ -105,6 +105,7 @@ I/O
 
 - Avoid calling ``S3File.s3`` when reading parquet, as this was removed in s3fs version 0.3.0 (:issue:`27756`)
 - Better error message when a negative header is passed in :func:`pandas.read_csv` (:issue:`27779`)
+- Added notes on Engine Disposal and user responsibility to multiple sql related function docstrings (:issue:`23086`)
```
Comment (gfyoung) [inline doc/source/whatsnew/v0.25.1.rst]: @TomAugspurger : Do we document entries for this?

Comment (TomAugspurger) [reply doc/source/whatsnew/v0.25.1.rst]: Yes.

File: pandas/io/sql.py (hunk 1 above)
Comment (poojahagawane) [reply pandas/io/sql.py]: Hey I added the newline however its not rendering in the documentation generated.

Comment (TomAugspurger) [reply pandas/io/sql.py]: I think they end up in the same paragraph in the generated HTML.

File: pandas/io/sql.py [hunk 3]
```diff
@@ -279,11 +287,15 @@ def read_sql_query(
     ----------
     sql : string SQL query or SQLAlchemy Selectable (select or text object)
         SQL query to be executed.
-    con : SQLAlchemy connectable(engine/connection), database string URI,
+    con : SQLAlchemy connectable(engine/connection) or database string URI
```
Comment (TomAugspurger) [inline pandas/io/sql.py]: Why this change?

File: doc/source/whatsnew/v0.25.1.rst [hunk 4]
```diff
@@ -72,7 +72,6 @@ I/O
 
 - Avoid calling ``S3File.s3`` when reading parquet, as this was removed in s3fs version 0.3.0 (:issue:`27756`)
 - Better error message when a negative header is passed in :func:`pandas.read_csv` (:issue:`27779`)
-- Follow the ``min_rows`` display option (introduced in v0.25.0) correctly in the HTML repr in the notebook (:issue:`27991`).
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.1.rst]: can you revert this file

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
can you merge master and revert the 0.25.1 file

**Comment by WillAyd** [comment]
Nice idea but looks like this has gone stale so closing; ping if you'd like to pick this back up
